Fixes cryptokitties loading/ordering

krzyzowiec opened the pull request:
[comment]: # (Please replace ... with your information. Remove < and >)
[comment]: # (To auto-close issue on merge, please insert the related issue number after # i.e fixes #566)

Fixes #6839 

### Summary

This is a completion of/improvement upon the following [work](https://github.com/status-im/status-react/pull/9810). The kitty ids are grabbed in groups of 20 and sent to :load-kitties to be loaded individually. The problem of ordering is not due to any strange behavior on the part of mapv, but due to the perfectly normal behavior of a plain hash map. It doesn't care about ordering.

This was a bit tricky to diagnose because the :CK collectible hash doesn't exist at db initialization, it is created when :load-collectible-success decides to merge the first kitty into the db under the :CK symbol and sees that it doesn't exist, after which merge creates a hash-map for the non-existent key. (this is the default behavior of merge)

To fix this, I update db in the :load-kitties handler to ensure that the :CK key is initialized to an empty ordered hash map, which I set to descending order. Here I am following the cryptokitties API, which returns the most recently created kitties first. (a custom function can be supplied to change the ordering if necessary, as I have no idea what order they use to display their cats on their site)

### Testing notes
Testing this in the app was a bit tricky. I loaded some kitties manually in the repl, but it would be good for someone to check this with actual kitties on their wallet account.

##### Functional

- 1-1 chats
- public chats
- group chats
- wallet / transactions
- dapps / app browsing
- account recovery
- new account
- user profile updates
- networks
- mailservers
- fleet
- bootnodes

##### Non-functional

- battery performance
- CPU performance / speed of the app
- network consumption

### Steps to test
<!-- (Specify exact steps to test if there are such) -->

Open status application
Navigate to the wallet tab
Expand a wallet with more than 20 kitties
Expand the collectibles tab and select kitties
Ideally see all kitties in the proper order

<!-- (PRs will only be accepted if squashed into single commit.) -->

status: ready <!-- Can be ready or wip -->

Commit:
fix #9903 hide buttons of actions that are not yet implemented

fix ens release button which showed a hardcoded date
link to release instructions on hackmd

Signed-off-by: yenda <eric@status.im>

jakubgs committed:
fastlane: disable oldest_build_allowed for TestFlight clean

Signed-off-by: Jakub Sokołowski <jakub@status.im>

Commit:
Disable editprofile tests

Signed-off-by: Serhy <sergii@status.im>

jinhojang6 committed:
update Italian, Arabic, Spanish(LatAm), Chinese and Korean

Signed-off-by: Andrey Shovkoplyas <motor4ik@gmail.com>

krzyzowiec committed:
Fixes cryptokitties loading/ordering

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :x: | a70fa29a | [#1](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-9939/1/) | 2020-01-29 02:59:05 | ~8 min | `ios` | [:page_facing_up:`log`](https://ci.status.im/job/status-react/job/prs/job/ios/job/PR-9939/1/consoleText) |

rasom committed:
[#9927] Fast blocks sync after being offline

Front end changes:
 As there is no guarantee that `newblock` event will be dispatched
consequently (e.g. if there was a delay after block#1 the next event
might be dispatched for a block#2000, no guarantee that block#2 will be
the next one), `newTransactions` field was added with a map of accounts
to the number of new transactions received to this block. In result if
there are new transactions we request them all of db instead of fetching
them on for the block specified in `newblock` event, as it was done
previously.

Back end changes:
- In order to avoid handling of the reorganized blocks we use an offset
from the latest known block when start listening to new blocks. Before
this commit the offset was 15 blocks for all networks. This offset is
too big for mainnet and causes noticeable delay of marking a transfer as
confirmed in Status (comparing to etherscan). So it was changed to be 5
blocks on mainnet and is still 15 blocks on other networks.
- Also before this commit all new blocks were handled one by one with
network specific interval (10s for mainnet), which means that in case of
lost internet connection or application suspension (happens on iOS)
receiving of new blocks would be paused and then resumed with the same
"speed" - 1 blocks per 10s. In case if that pause is big enough the
application would never catch up with the latest block in the network,
and this also causes the state of transfers to be delayed in the
application. In this commit in case if there was more than 40s delay
after receiving of the previous block the whole history in range between
the previous received block and ("latest"-reorgeSafetyDepth) block is
checked at once and app catches up with a recent state of the chain.

Ferossgp committed:
Remove contact indicator from userpic

Add contact indicator in 1:1 as a text. Fixes #9826

Fix e2e

explicitly define line height

On android and iOs line height is different, explicitly define so it will be the same in both platforms.

Signed-off-by: Gheorghe Pinzaru <feross95@gmail.com>

flexsurfer commented on src/status_im/wallet/collectibles/core.cljs:
```diff
@@ -54,7 +54,7 @@
  :load-kitties
  (fn [{db :db} [_ ids]]
-   {:db db
+   {:db (update-in db [:collectibles] merge {ck (sorted-map-by >)})
```
could you elaborate on why we need this? thanks

krzyzowiec replied:
The original bug report mentioned that the displayed order of the cats is different than on the website. That's because they are unsorted.

If you mean why that code in that precise spot, it's because you must have the sorted map there first, before you add any cats, if you want to maintain a sorted order after the merges. Putting it there seemed like the least obtrusive way to do it. (I don't have to touch the initialization, and every other function stays the same)

Did you try it and compare?

cammellos committed:
Explicitly start messenger

We should only start receiving messages/processing topics once all the
initializiation is completed, so an endpoint that start the messenger
has been added.
Before we would initialize the topics/receiving messages at the same time,
which resulted in a race condition where a topic was signaled from
status-go and was considered new just because was not loaded yet.

Signed-off-by: Andrea Maria Piana <andrea.maria.piana@gmail.com>

Commit:
fix hardcoded value in spanish translation

Signed-off-by: Andrey Shovkoplyas <motor4ik@gmail.com>

flexsurfer committed:
[#9954] Add 'view profile' to message long tap

Signed-off-by: yenda <eric@status.im>

Ferossgp committed:
Git ignore trace.edn and config.cljs

They are auto generated and should not be present in git

Signed-off-by: Gheorghe Pinzaru <feross95@gmail.com>

rasom committed:
[#9942] Upgradable paths in configs

Storing absolute path for different configs breaks compatibility on iOS
as app's dir is changed after upgrade. The solution is to store relative
paths and to concatenate it with `backend.rootDataDir`. The only
exception is `LogFile` as it is stored outside `backend.rootDataDir` on
Android. `LogDir` config was added to allow adding of custom dir for log
file.
Configs concerned:
`DataDir`
`LogDir`
`LogFile`
`KeystoreDir`
`BackupDisabledDataDir`

flexsurfer committed:
remove event duplication

Signed-off-by: yenda <eric@status.im>

flexsurfer committed:
load more ui fix

Signed-off-by: Andrey Shovkoplyas <motor4ik@gmail.com>

Serhy commented:
Hey @krzyzowiec , builds failed due to incorrect formatting (https://ci.status.im/blue/organizations/jenkins/status-react%2Fprs%2Fandroid/detail/PR-9939/1/pipeline )
Could you look at it, please?

krzyzowiec committed:
Merge branch 'develop' of https://github.com/krzyzowiec/status-react into develop
